resource/ui/ts/controllers/rest_explorer.ts

```diff
@@ -1,53 +1,306 @@
 // source: controllers/rest_explorer.ts
-/// <reference path="../typings/angularjs/angular.d.ts" />
-
-// Author: Andrew Bonventre (andybons@gmail.com)
-var crApp = angular.module('cockroach');
-crApp.controller('RestExplorerCtrl', ['$scope', '$http', function(scope, http) {
-  scope.kvCounterVal = 0;
-  scope.responseLog = [];
-  scope.clearResponseLog = function(e) {
-    scope.responseLog = [];
-  };
-  scope.requestPending = false;
-  scope.handleClick = function(e) {
-    e.preventDefault();
-    var method = e.target.getAttribute('data-method');
-    var endpoint = e.target.getAttribute('data-endpoint');
-    if (endpoint == '/kv/rest/range') {
-      endpoint += '?start=' + encodeURIComponent(scope.kvRangeStart);
-      if (!!scope.kvRangeEnd) {
-        endpoint += '&end=' + encodeURIComponent(scope.kvRangeEnd);
-      }
-    } else if (!!scope.kvKey) {
-      endpoint += scope.kvKey;
-    }
-    var data = '';
-    if (!!scope.kvValue) {
-      data = scope.kvValue;
-    }
-    var req = {
-      method: method,
-      url: endpoint,
-      headers: {
-        'Content-Type': 'text/plain; charset=UTF-8'
-      },
-      data: data
-    };
-    if (endpoint.indexOf('/kv/rest/counter/') != -1) {
-      req.headers = {
-        'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8'
+/// <reference path="../typings/mithriljs/mithril.d.ts" />
```

Comment:
Mithril seems a lot more verbose than angular it's replacing. It can't just be typescript. Does that worry you?